### PR TITLE
Add 'isXYZ' functions

### PR DESCRIPTION
**Pull request by joaolucasl**
This PR adds the following files and its tests:
```
isFinite
isFunction
isInteger
isNan  
isNumber
isObject
```

### CODE REVIEW

File: test/internals/is-number.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+test('returns true for a primitive number', t => {
+  t.true(isNumber(42))
+})
+
+test('returns true for Object-constructed numbers ', t => {
+  t.true(isNumber(Object(42)))
+})
+
+test('returns true for NaN ', t => {
+  t.true(isNumber(NaN))
```
Comment (joaolucasl) [inline test/internals/is-number.js:13]: lol Javascript

Comment (grvcoelho) [reply test/internals/is-number.js:13]: :smile:

File: - [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Uso: grep [OPÇÃO]... PADRÃO [ARQUIVO]...
```
Comment (grvcoelho) [inline -]: lol, hahahaha @joaolucasl

Comment (joaolucasl) [reply -]: lol wat

File: lib/internals/is-finite.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import _isNaN from './is-nan'
```
Comment (grvcoelho) [inline lib/internals/is-finite.js:1]: Why is everything with `_` before their names?

haha I know you probably mean private, but we could leave them off. :smile:

**Review by grvcoelho — CHANGES_REQUESTED** [review]

File: lib/internals/is-finite.js (hunk 3 above)
Comment (joaolucasl) [reply lib/internals/is-finite.js:1]: So it doesn't override [JS's default `isNaN`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/isNaN) function

File: lib/internals/is-finite.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+import _isNumber from './is-number'
+
+const _isFinite = x => !(!_isNumber(x) || _isNaN(x) || x === Infinity || x === -Infinity)
```
Comment (joaolucasl) [inline lib/internals/is-finite.js]: Same for `isFinite`, there's a [default JS function for that](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Global_Objects/isFinite).

`isNumber` has `_` for consistency

File: lib/internals/is-function.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const _typeofIsFunction = x => (typeof x === 'function')
+const _toStringIsFunction = x => (Object.prototype.toString.call(x) === '[object Function]')
+const _toStringIsGenerator = x => (Object.prototype.toString.call(x) === '[object GeneratorFunction]')
```
Comment (joaolucasl) [inline lib/internals/is-function.js]: These have `_` for no actual reason, mostly being private (as in, not exposed)

**Comment by joaolucasl** [comment]
Finally had the time to update this @grvcoelho. Mind givin' it a check?

**Review by grvcoelho — APPROVED** [review]

**Pull request merged**